src/test/java/ICS_Footer.java

```diff
@@ -0,0 +1,230 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class ICS_Footer {
+    public static void main(String[] args) throws InterruptedException {
+        footerTest("desktop");
+        footerTest("tablet");
+        footerTest("mobile");
+    }
+
+    public static void footerTest(String size) throws InterruptedException {
+        WebDriverManager.chromedriver().setup();
+
+        WebDriver driver = driverSetup(size);
+
+        Dimension desktop = new Dimension(1366, 766);
+        Dimension tablet = new Dimension(766, 1024);
+        Dimension mobile = new Dimension(400, 800);
+        if (size.equals("desktop")){
+            driver.manage().window().setSize(desktop);
+        }else if (size.equals("tablet")){
+            driver.manage().window().setSize(tablet);
+        }else{
+            driver.manage().window().setSize(mobile);
+        }
+
+        driver.get("https://icecreamshop-frontend-rkbemzbp2q-ez.a.run.app/");
+
+        //fetching footer element
+        WebElement footer = driver.findElement(By.className("footer"));
+
+        //first footer
+        WebElement footer1 = footer.findElement(By.cssSelector("[class^='footer-']"));
+
+        //fetching the column title elements into a list
+        List<WebElement> columnList = footer1.findElements(By.cssSelector("h4"));
+
+        //checking the number of columns
+        Assert.assertEquals(4, columnList.size());
+
+        //checking column data
+        if (size.equals("desktop")){
+            columnDataTest(columnList, size);
+        }else if (size.equals("tablet")){
+            columnDataTest(columnList, size);
+        }else{
+            columnDataTest(columnList, size);
+        }
+
+        //fetching the row sub-elements into a list
+        List<WebElement> rowList = footer1.findElements(By.cssSelector("a"));
+
+        //checking row data
+        if (size.equals("desktop")){
+            rowListTest(rowList, size, footer1);
+        }else if (size.equals("tablet")){
+            rowListTest(rowList, size, footer1);
+        }else{
+            rowListTest(rowList, size, footer1);
+        }
+
+        //second footer
+        WebElement footer2 = footer.findElement(By.className("footer1"));
+
+        WebElement title = footer2.findElement(By.cssSelector("h1"));
+        String titleText = title.getText();
+
+        Assert.assertEquals("@2021 Gelato & Donuts.", titleText);
+
+
+        //newsletter text field element
+        WebElement newsletterTextField = footer1.findElement(By.cssSelector("[id$='-input']"));
+
+        //newsletter button element
+        WebElement newsletterButton = footer1.findElement(By.cssSelector("[id$='-button']"));
+
+        //checking email validation - good email
+        newsletterTextField.click();
+        newsletterTextField.sendKeys("test@test");
+
+        Thread.sleep(1000);
```

Comment:
FOR waits : 
This is an option, but usually when selenium clicks a lot of elements you might want to use something like this in the future : 

        WebDriverWait wait = new WebDriverWait(driver, 10);  // depending on the load time you just use a bigger value
        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector('LOCATOR')));
/**Also another condition is this one : ExpectedConditions.elementToBeClickable. */